### DISCUSSION TITLE
Resources pages component ?

### DISCUSSION THREAD

**Discussion by VirginieLemaire** [danielschuster-muc/potter-db#762] [Ideas]
I noticed that resources pages have the same layout:
- title
- introduction sentence
- attributes title
- table mapping an array of attributes

It triggered a "factoring alert" in my head 😅 

💡 Nextra seems pretty cool using `mdx` which allows us to use `jsx` .
It seems possible to create a component so translators' job would be more effective (less repetitions).

I imagine something like this:
- 1 json file with the attributes datas (e.g. `books.en.json`)
- 1 page for the resource using the json file and the component (e.g. `books.en.mdx`)

The component could have 3 or 4 props:
- title: the translated title
- sentence: the translated sentence introducing the resource
- attribute: the translated attribute
- the table mapping on the json array

The books page could be something like

```mdx
import books from "./datas/books.en.json"; // or another folder name
import Resource from "../someFolderForDocComponents/Resource.jsx";

<Resource
  title={"Books"}
  sentence={"This resource provides information about books from the Harry Potter Universe."}
  attribute={"Attributes"}
  data={books}
/>
```
or like that

```mdx
import books from "./datas/books.en.json"; // or another folder name
import Attributes from "../someFolderForDocComponents/Attributes.jsx";

# Books

This resource provides information about books from the Harry Potter Universe.

## Attributes

<Attributes
  data={books}
/>
```

Translators could translate datas separately and would only have to change some words in the pages (inside the component props).
This would also open the possibility to "cut" translation tasks into "folders" if a contributor wants to help but have not many time for this 🤔 

What do you think ? 
If it's too much (or complicating things for no real advantages), no worries, you can tell me 😅 . It's just an idea

**Comment by danielschuster-muc** [comment]
Hi there, 
thanks for you idea. It always nice if people come up with some ideas to help / improve Potter DB. 
However I am not quite sure how to implement this idea and how much of an advantage it is, especially when multiple people are currently translating the documentation. Of course it is always good to do refactoring, but I am not sure whether this simplifies the translating process. 

I would however still prefer to have the translations in the `.mdx` files and not using external `.json` files. 
Maybe you could try some things out, maybe make a draft pr and then we could further discuss it? 

Thanks for you contribution / sauggestion.

**Reply by VirginieLemaire** [reply]
Yes, you're right, for the moment many people are working on translating, it would be too much work for them.
As you prefer having translations in the `.mdx` file, this is not relevant to make this refacto.

I may try, just for fun, after Hacktobertfest.
As you suggested, I might make a draft PR (would be the 1st time, another occasion to try something new 🤩 ) when I'll "play with this idea" (without waiting for implementation).
